### DISCUSSION TITLE
如何拓展webpack css-loader/less-loader配置？

### DISCUSSION THREAD

**Discussion by MatrixAge** [umijs/umi#9741] [Q&A] [answered]
https://github.com/Wildhoney/ReactShadow

使用了一个react-shadow用来在react中使用shadow dom，但是shadow dom的css需要使用内联样式，内联样式纯拼字符串还是比较反人类的，官方提供了一种方式也就是使用to-string-loader来引入style：

```tsx
import root from 'react-shadow';
import styles from './styles.css';

export default function Quote() {
    return (
        <root.div className="quote">
            <q>There is strong shadow where there is much light.</q>
            <span className="author">― Johann Wolfgang von Goethe.</span>
            <style type="text/css">{styles}</style>
        </root.div>
    );
}
```

```js
{
    test: /\.css$/,
    loader: ['to-string-loader', 'css-loader']
}
```

不知umi4如何输出上述配置（css-loader & less-loader）？

**Comment by fz6m** [comment, marked as answer]
用 css in js 吧，调整 css 或 less 文件的加载方式会影响到其他使用位置，官方示例中的做法会污染全局，不适合混用的场景。

**Reply by MatrixAge** [reply]
现阶段的css in js太反人类了，没啥提升，还多出很多内存占用。

tss-react算是能用的方案

**Reply by dslian** [reply]
使用inline-loader

**Reply by MatrixAge** [reply]
经过尝试之后，解决方案是：自定义一种格式比如.sss, 然后用to-string-loader和less/scss-loader，在vscode里把.sss映射成.less/scss，就能配合react-shadow畅快地使用shadow dom写组件了。
